src/HDF5Msg.chpl

```diff
@@ -896,6 +885,186 @@ module HDF5Msg {
+                var localSegs = new lowLevelLocalizingSlice(segEntry.a, 0..#segEntry.size);
+
+                select dType {
+                    when (DType.Int64) {
+                        var valEntry:SymEntry = toSymEntry(toGenSymEntry(vals), int);
+
+                        //localize values and write dataset
+                        var localVals = new lowLevelLocalizingSlice(valEntry.a, 0..#valEntry.size);
+                        writeLocalSegArray(file_id, group, localVals, valEntry.size, localSegs, segEntry.size, int);
+                        dtype = getDataType(int);
+                    } when (DType.UInt64) {
+                        var valEntry:SymEntry = toSymEntry(toGenSymEntry(vals), uint);
+
+                         //localize values and write dataset
+                        var localVals = new lowLevelLocalizingSlice(valEntry.a, 0..#valEntry.size);
+                        writeLocalSegArray(file_id, group, localVals, valEntry.size, localSegs, segEntry.size, uint);
+                        dtype = getDataType(uint);
+                    } when (DType.Float64) {
+                        var valEntry:SymEntry = toSymEntry(toGenSymEntry(vals), real);
+
+                         //localize values and write dataset
+                        var localVals = new lowLevelLocalizingSlice(valEntry.a, 0..#valEntry.size);
+                        writeLocalSegArray(file_id, group, localVals, valEntry.size, localSegs, segEntry.size, real);
+                        dtype = getDataType(real);
+                    } when (DType.Bool) {
+                        var valEntry:SymEntry = toSymEntry(toGenSymEntry(vals), bool);
+
+                         //localize values and write dataset
+                        var localVals = new lowLevelLocalizingSlice(valEntry.a, 0..#valEntry.size);
+                        writeLocalSegArray(file_id, group, localVals, valEntry.size, localSegs, segEntry.size, bool);
+                        dtype = getDataType(bool);
+                    }
+                    otherwise {
+                        throw getErrorWithContext(
+                           msg="Unsupported SegArray DType %s".format(dtype2str(dType)),
+                           lineNumber=getLineNumber(),
+                           routineName=getRoutineName(), 
+                           moduleName=getModuleName(),
+                           errorClass="IllegalArgumentError");
+                    }
+                }
+                
+                writeArkoudaMetaData(file_id, group, objType, dtype);
+                C_HDF5.H5Fclose(file_id);
+            }
+            when MULTI_FILE {
+                var filenames = prepFiles(filename, mode, segEntry.a);
+
+                const lastSegIdx = segEntry.a.domain.high;
+
+                // helper function definition
+                inline proc writeSegmentedDistDset(filenames: [] string, group: string, objType: string, values, segments, st: borrowed SymTab, type t) throws {
```

Comment:
Again, I think we should pull this function out to the module level to match structure used elsewhere.

Reply:
Function moved to module level